Source/DafnyDriver/Commands/MeasureComplexityCommand.cs

```diff
@@ -75,13 +84,42 @@ static class MeasureComplexityCommand {
     return await compilation.GetAndReportExitCode();
   }
 
+  public static async Task ReportResourceSummary(
+    CliCompilation cliCompilation,
+    IObservable<CanVerifyResult> verificationResults) {
+
+    PriorityQueue<VerificationTaskResult, int> worstPerformers = new();
+
+    var topX = cliCompilation.Options.Get(TopX);
+    verificationResults.Subscribe(result => {
+      foreach (var taskResult in result.Results) {
+        var runResult = taskResult.Result;
+        worstPerformers.Enqueue(taskResult, runResult.ResourceCount);
+        if (worstPerformers.Count > topX) {
+          worstPerformers.Dequeue();
+        }
+      }
+    });
+    await verificationResults.WaitForComplete();
+    var output = cliCompilation.Options.OutputWriter;
+    var decreasingWorst = new Stack<VerificationTaskResult>();
+    while (worstPerformers.Count > 0) {
+      decreasingWorst.Push(worstPerformers.Dequeue());
+    }
+
+    foreach (var performer in decreasingWorst) {
+      await output.WriteLineAsync($"Verification task on line {performer.Task.Token.line} in file {performer.Task.Token.filename} consumed {performer.Result.ResourceCount} resources");
```

Comment:
This is a really long message to parse. I'd prefer something like `file.dfy(9): <some description of context, maybe function/method/lemma name> used 4389 RU". If you leave it as-is, the term "resources" is kind of vague. I'd maybe use "RU" to match what the IDE does?

Reply:
What do you consider long? This generally fits within 80 characters. Is there a reason for things to be shorter? Also having the line number in there is a must. We will move towards `--isolate-assertions` being the default, so having a function/method/lemma name won't say much.

I'm not a fan of abbreviations in user interfaces. `resources` is understandable for someone who has not read the manual, while `RU` is not. `resource units` as opposed to `resources` does not give me any extra information. Alternatively we could call it complexity. Then it would say "Verification task... has complexity {...}", which I think would be more intuitive than resources.